### ISSUE TITLE
Reference alleles selection

### ISSUE THREAD

**Issue by saramonzon** [BU-ISCIII/taranis#12]
The idea is to cluster the alleles present in the database for each gene by sequence similarity without previous knowledge of the number of clusters. We can use a sequence identity threshold of 90% (or the equivalent of mash distance), or any other we can evaluate.

**Option 1:**
Use mash distance and some clustering algorithm
**Option 2:**
Use vsearch, cd-hit, etc
**Option 3:** 
Use [ALFAClust](https://bmcbioinformatics.biomedcentral.com/articles/10.1186/s12859-022-04643-9)

**Comment by luissian** [comment]
I have been testing ALFAClust with 1 allele in the schema. It is really fast in writing the  result.  Using the parameter "-l 0.925" to get an average estimated of 0.925, the alleles were classified in 21 clusters.
Output result is a file with the cluster name followed with the locus name of this cluster, then next line the next cluster with the locus. See the example bellow:

`#Cluster 1
lmo0002_36
lmo0002_79
lmo0002_90

.....
....

#Cluster 2
lmo0002_98
lmo0002_101
 and so on
`

By grepping the output file searching for "#Cluster" and -n to get the number lines, we get something like this.
`
1:#Cluster 1
78:#Cluster 2
81:#Cluster 3
795:#Cluster 4

`
Meaning that locus name from line 2 up to 77 belongs to Cluster 1. 
I am guessing (maybe I am wrong) that I could take  any of this locus, randomly,  to  map all the locus on the cluster with an identity of 90% , using blast. 
Unfortunately, I did not have time to verify it, yet. 
If I am right we could use this alternative because of the fast output

**Comment by luissian** [comment]
I did more  testing, based on the clustered file generated by alfaclust.
I selected some cluster randomize and for each cluster a random allele was selected.
Then using blast and the random allele,  I get that every time run the script for checking the number of alleles that blast matches wit a 90% of identity I get that all alleles were included in the match.
However the first time running for testing, I select the last allele in the first group of cluster, I get only 71 of 76  matches. But  it was the only time that get a value bellow 100% of matches.
To be sure we could run blast each cluster to verify that the selected random allele matches all alleles in the cluster.

I hope this test was useful for  making the final decision.
